src/app/settings/views/Configuration/CommissioningForm/CommissioningForm.test.tsx

```diff
@@ -77,13 +77,22 @@ describe("CommissioningForm", () => {
         </MemoryRouter>
       </Provider>
     );
-    submitFormikForm(wrapper, {
-      commissioning_distro_series: "bionic",
-      default_min_hwe_kernel: "ga-16.04-lowlatency",
-      maas_auto_ipmi_user: "maas",
-      maas_auto_ipmi_k_g_bmc_key: "password",
-      maas_auto_ipmi_user_privilege_level: "USER",
+
+    const maas_auto_ipmi_user_input = screen.getByRole("textbox", {
+      name: "MAAS generated IPMI username",
     });
+    await userEvent.clear(maas_auto_ipmi_user_input);
+    await userEvent.type(maas_auto_ipmi_user_input, "maas");
+
+    const maas_auto_ipmi_k_g_bmc_key_input = screen.getByLabelText(
+      "K_g BMC key textbox"
+    );
```

Comment:
Could this use the role as well? You might need to use a regex for the name as the label also contains other elements.

`screen.getByRole("textbox", { name: name: new RegExp(Label.ipmiKGBmcKey) });``

where the label would be something like:

```typescript
enum Label {
    ipmiKGBmcKey = "K_g BMC key"
}

```

Here's an example of us doing something similar: https://github.com/canonical-web-and-design/maas-ui/blob/main/src/app/machines/components/MachineHeaderForms/ActionFormWrapper/TagForm/TagFormChanges/TagFormChanges.test.tsx#L88

Reply:
I'm not sure this is possible, since when the label renders, it doesn't actually appear to be attached to the input. I tried using the regex but unfortunately had no luck

```html
<label
    class="p-form__label"
    for="mock-redux-js-nanoid-4"
>
    K_g BMC key
    
    <span>
    <span
        style="display: inline-block;"
    >
        <button
        aria-describedby="mock-nanoid-9"
        class="p-button--base has-icon is-small u-no-border u-no-line-height u-no-margin"
        type="button"
        >
        <i
            class="p-icon--help"
        />
        </button>
    </span>
    </span>
</label>
<div
    class="p-form__control u-clearfix"
>
    <input
    aria-describedby="mock-nanoid-8"
    aria-invalid="false"
    aria-label="K_g BMC key"
    autocomplete="new-password"
    class="p-form-validation__input"
    id="mock-redux-js-nanoid-4"
    label="[object Object]"
    name="maas_auto_ipmi_k_g_bmc_key"
    type="password"
    value=""
    />
```

Reply:
Ah sorry, I missed that this was a password field so I think from memory it behaves a bit differently.